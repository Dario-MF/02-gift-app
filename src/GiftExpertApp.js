import React, { useState } from 'react'
import AddCategory from './Components/AddCategory'
import GridCategory from './Components/GridCategory'

const GiftExpertApp = ( { defaultCategories = []}) => {


    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>GiftExpertApp</h2>
            <hr></hr>
            <AddCategory setCategories={setCategories} />
            <ol>
                {
                    categories.map(category => (
                        <GridCategory category={category} key={category} />
                    ))
                }
            </ol>
        </>
    )
}

export default GiftExpertApp
