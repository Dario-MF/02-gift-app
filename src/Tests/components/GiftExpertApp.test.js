import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GiftExpertApp from '../../GiftExpertApp'


describe('Test for <GiftExpertApp />', () => {
    
    test('snapshot for <GiftExpertApp />', () => {
        const wrapper = shallow(<GiftExpertApp  defaultCategories={['One Piece']}/>)

        expect(wrapper).toMatchSnapshot()
    })
    test('evaluar que se impriman las categorias', () => {
        const categories = ['one Punch', 'Dragon Ball']
        const wrapper = shallow(<GiftExpertApp defaultCategories={ categories } />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GridCategory').length).toBe(categories.length)

    })
    
    
})
