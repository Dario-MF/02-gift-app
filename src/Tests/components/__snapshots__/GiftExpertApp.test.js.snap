// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GiftExpertApp /> evaluar que se impriman las categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GridCategory
      category="one Punch"
      key="one Punch"
    />
    <GridCategory
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test for <GiftExpertApp /> snapshot for <GiftExpertApp /> 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GridCategory
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
